### PR TITLE
Adapt to changes in libflatsurf 2.0.0

### PR DESCRIPTION
**Pull request by saraedum**
and speed up tests so that they do not take 3 minutes on my machine
anymore…

### CODE REVIEW

File: flatsurf/geometry/gl2r_orbit_closure.py [hunk 1]
```diff
@@ -988,41 +989,28 @@ def decomposition(self, v, limit=-1):
...
+            connections = connections.byLength()
 
-        visited[slope].append((x, y))
+        Vector = cppyy.gbl.flatsurf.Vector[type(self._surface).Coordinate]
+        slopes = cppyy.gbl.std.set[Vector, Vector.CompareSlope]()
 
-        return True
+        for connection in connections:
+            direction = connection.vector()
+            if slopes.find(direction) != slopes.end():
```
Comment (saraedum) [inline flatsurf/geometry/gl2r_orbit_closure.py:1004]: We cannot say `direction in slopes` due to https://bitbucket.org/wlav/cppyy/issues/278/membership-test-in-std-set-is-broken-slow.

File: flatsurf/geometry/gl2r_orbit_closure.py [hunk 2]
```diff
@@ -657,7 +654,9 @@ def field_of_definition(self):
             GL(2,R)-orbit closure of dimension at least 4 in H_7(4^3, 0) (ambient dimension 17)
             sage: O.field_of_definition() # optional: exactreal, pyflatsurf
             Number Field in c0 with defining polynomial x^2 - 2 with c0 = 1.414213562373095?
+            sage: bound = E.billiard_unfolding_stratum('half-translation', marked_points=True).dimension()
             sage: for decomposition in O.decompositions(1):  # long time, optional: exactreal, pyflatsurf
+            ....:     if O.dimension() == bound: break
```
Comment (videlec) [inline flatsurf/geometry/gl2r_orbit_closure.py:659]: Don't do that (or at least don't do that everywhere). We want to check that all found directions are contained in the target space and we don't get something bigger than expected.

File: flatsurf/geometry/gl2r_orbit_closure.py [hunk 3]
```diff
@@ -669,7 +668,9 @@ def field_of_definition(self):
             sage: O = GL2ROrbitClosure(S) # optional: pyflatsurf
             sage: O.field_of_definition() # optional: pyflatsurf
             Number Field in c0 with defining polynomial x^3 - 3*x - 1 with c0 = 1.879385241571817?
-            sage: for decomposition in O.decompositions(1):  # long time, optional: pyflatsurf
+            sage: bound = E.billiard_unfolding_stratum('half-translation', marked_points=True).dimension()
+            sage: for decomposition in O.decompositions(1):  # long time, optional: exactreal, pyflatsurf
+            ....:     if O.dimension() == bound: break
```
Comment (videlec) [inline flatsurf/geometry/gl2r_orbit_closure.py:673]: similar

File: flatsurf/geometry/gl2r_orbit_closure.py (hunk 2 above)
Comment (saraedum) [reply flatsurf/geometry/gl2r_orbit_closure.py:659]: Yes, I agree. But we need to break after a few iterations extra iterations at least. Doctests that run for several minutes (on a single thread) are not useful anymore even if marked with long time imho.

Comment (videlec) [reply flatsurf/geometry/gl2r_orbit_closure.py:659]: What about using `islice` from itertools to bound the number of saddle connection used?
```
from itertools import islice
for decomposition in islice(O.decompositions(1), 10):  # long time, optional: pyflatsurf
    O.update_tangent_space_from_flow_decomposition(decomposition)
    print(O.dimension())
```

Comment (videlec) [reply flatsurf/geometry/gl2r_orbit_closure.py:659]: Though, these doctests are intended for being useful for the user. If someone wants to compute orbit closure of her prefered surface, she might prefer to see your version.

**Pull request merged**